velox/external/hdfs/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if(${VELOX_ENABLE_HDFS})
+  velox_add_library(velox_external_hdfs ArrowHdfsInternal.cpp)
+  velox_link_libraries(
+        velox_external_hdfs
+        PRIVATE
+        arrow)
```

Comment:
why use `PRIVATE`? 
If arrow has already installed in non-system path, and be included with `CMAKE_PREFIX_PATH`, `velox_hdfs` target can not found arrow header files.

```
# PRIVATE
/usr/lib64/ccache/c++ -I/root/oap-velox/. -I/root/oap-velox/velox/external/xxhash -isystem /root/oap-velox/velox -isystem /root/oap-velox/velox/external -isystem /usr/include/libdwarf -isystem /root/oap-velox/velox/external/date/velox/external  -o velox/connectors/hive/storage_adapters/hdfs/CMakeFiles/velox_hdfs.dir/HdfsReadFile.cpp.o -c /root/oap-velox/velox/connectors/hive/storage_adapters/hdfs/HdfsReadFile.cpp

# NON-PRIVATE
/usr/lib64/ccache/c++ -I/root/oap-velox/. -I/root/oap-velox/velox/external/xxhash -isystem /root/oap-velox/velox -isystem /root/oap-velox/velox/external -isystem /opt/extra_lib/arrow_ep/include -isystem /root/apache-arrow/cpp/_build/thrift_ep-install/include -isystem /usr/include/libdwarf -isystem /root/oap-velox/velox/external/date/velox/external -o velox/connectors/hive/storage_adapters/hdfs/CMakeFiles/velox_hdfs.dir/HdfsReadFile.cpp.o -c /root/oap-velox/velox/connectors/hive/storage_adapters/hdfs/HdfsReadFile.cpp
```

@JkSelf @PHILO-HE

Reply:
@Yohahaha, thanks for raising up this issue. The PRIVATE linkage will make the headers bound to target arrow not accessible to target `velox_external_hdfs`. If those headers are in system path, it is not an issue since compiler can always find them from system path. Otherwise, build error will occur. @JkSelf, if there is no special reason to use PRIVATE, could you change it to PUBLIC?

Reply:
@Yohahaha @PHILO-HE Yes. I will file a pr to change to `PUBLIC`. Thanks.

Reply:
Sorry I did not catch this. I saw the arrow headers in the implementation file and didn't check the header as I usually only include things that are used privately in the implementation and everything else in the header.